wetterdienst/core/scalar/interpolate.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018-2022, earthobservations developers.
+# Distributed under the MIT License. See LICENSE for more info.
+import logging
+from datetime import datetime
+from functools import lru_cache
+from itertools import combinations
+from queue import Queue
+
+import numpy as np
+import pandas as pd
+import utm
+from scipy import interpolate
+from shapely.geometry import Point, Polygon
+
+from wetterdienst.metadata.columns import Columns
+
+log = logging.getLogger(__name__)
+
+
+class _ParameterData:
+    def __init__(self, values: pd.DataFrame, station_ids: list = None, extra_station_counter=0):
+        self.station_ids = station_ids if station_ids else []
+        self.extra_station_counter = extra_station_counter
+        self.values = values
+        self.finished = False
+
+
+def get_interpolated_df(
+    stations_ranked, requested_x, requested_y, parameters, interpolatable_parameters
+) -> pd.DataFrame:
+    stations_dict, param_dict = request_stations(stations_ranked, interpolatable_parameters, parameters)
+    df = calculate_interpolation(requested_x, requested_y, stations_dict, param_dict)
+    df[Columns.DISTANCE_MEAN.value] = pd.Series(df[Columns.DISTANCE_MEAN.value].values, dtype=float)
+    df[Columns.VALUE.value] = pd.Series(df[Columns.VALUE.value].values, dtype=float)
+    df[Columns.DATE.value] = pd.to_datetime(df[Columns.DATE.value], utc=True)
```

Comment:
This is an interesting point here. I don't want to open a whole other topic, but what if we are interpolating on a weather service that provides data across different time zones e.g. portugal and spain?

I wouldn't automatically assume the timezone to be utc but the one of the point of interest.

Another point would be to make sure that the data coming from different time zones is first converted to the same timezone before doing other stuff with it.

This brings me to the idea: Should we enable kind of a beta mode where interpolation only works for DWD data?